src/Scripts/Game/Feature/Quantity/EL_QuantityComponent.c

```diff
@@ -0,0 +1,261 @@
+			quantitySource.AddQuantity(-transferQuantity);
+		}
+		else
+		{
+			// Source transferred entire quantity, so delete it
+			SCR_EntityHelper.DeleteEntityAndChildren(quantitySource.GetOwner());
+			return 0;
+		}
+
+		return quantitySource.GetQuantity();
+	}
+
+	//------------------------------------------------------------------------------------------------
+	void Split(int splitSize = -1)
+	{
+		if (splitSize == -1) splitSize = m_iQuantity / 2;
+
+		IEntity owner = GetOwner();
+		IEntity destinationEntity = EL_Utils.SpawnEntityPrefab(EL_Utils.GetPrefabName(owner), owner.GetOrigin());
+		EL_QuantityComponent quantityDestination = EL_ComponentFinder<EL_QuantityComponent>.Find(destinationEntity);
+		if (!quantityDestination) return;
+
+		quantityDestination.Combine(this, splitSize, true);
+
+		SetTransferIntent(destinationEntity, true);
+
+		InventoryItemComponent sourceInventoryItem = EL_ComponentFinder<InventoryItemComponent>.Find(owner);
+
+		// Ground item, just move it somewhere else so it can be picked up seperatly from the source stack
+		if (!sourceInventoryItem.GetParentSlot())
+		{
+			vector maxDims;
+			owner.GetBounds(null, maxDims);
+			float minRadius = Math.Max(maxDims[0], maxDims[2]); //Bounding radius
+			RandomGenerator random();
+			destinationEntity.SetOrigin(random.GenerateRandomPointInRadius(minRadius + 0.05, minRadius + 0.25, owner.GetOrigin()));
+			return;
+		}
+
+		// TODO: Refactor to use responsible storage manager if owned, if not owned always global one for TryInsertItemInStorage. No rpcs needed anymore. (likely 0.9.6 or 0.9.7 patch)
+		BaseInventoryStorageComponent storage = sourceInventoryItem.GetParentSlot().GetStorage();
+		InventoryStorageManagerComponent storageManager = EL_InventoryUtils.GetResponsibleStorageManager(owner);
+		if (!storageManager) storageManager = EL_GlobalInventoryStorageManagerComponent.GetInstance();
+
+		RplComponent replication = EL_ComponentFinder<RplComponent>.Find(storageManager.GetOwner());
+		if (replication.IsOwner())
+		{
+			storageManager.TryInsertItemInStorage(destinationEntity, storage);
+		}
+		else
+		{
+			if (storageManager.Type().IsInherited(SCR_InventoryStorageManagerComponent))
+			{
+				SCR_InventoryStorageManagerComponent.Cast(storageManager).EL_RequestInsertItemInStorage(destinationEntity, storage);
+			}
+			else if (storageManager.Type().IsInherited(SCR_VehicleInventoryStorageManagerComponent))
+			{
+				SCR_VehicleInventoryStorageManagerComponent.Cast(storageManager).EL_RequestInsertItemInStorage(destinationEntity, storage);
+			}
+			else
+			{
+				Debug.Error(string.Format("Unknown storage manager type: '%1'", storage.Type().ToString()));
+			}
+		}
+	}
+
+	//------------------------------------------------------------------------------------------------
+	static void SetTransferIntent(notnull IEntity sourceEntity, bool keepSeperate)
+	{
+		if (!s_mQuantityTransferIntents) s_mQuantityTransferIntents = new map<IEntity, bool>();
```

Comment:
A lot of long single line ifs again.

Reply:
This is not really long, anything that does not exceed the ---- width is usually ok I'd say